compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -92,6 +91,42 @@ impl<'a, 'tcx> Visitor<'tcx> for InferBorrowKindVisitor<'a, 'tcx> {
 }
 
 impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
+    fn log_new_capture_analysis(
+        &self,
+        closure_def_id: rustc_hir::def_id::DefId,
+        capture_information: &FxIndexMap::<Place<'tcx>, ty::CaptureInfo<'tcx>>
+    ) {
+        if self.should_log_capture_analysis(closure_def_id) {
+            for (place, capture_info) in capture_information {
+                let variable_name = match place.base {
```

Comment:
Also turn line 101-125 into sort of a closure/ separate function that takes a place and capture_information and returns the `{}[{}] -> {}", variable_name, projections_str, capture_kind_str)` part.

This way it can reused with logging min capture.